portal-bridge/src/beacon_bridge.rs

```diff
@@ -298,21 +311,41 @@ impl BeaconBridge {
     async fn serve_light_client_finality_update(
         api: ConsensusApi,
         portal_clients: Arc<Vec<HttpClient>>,
-    ) -> anyhow::Result<()> {
+        finalized_slot: u64,
+    ) -> anyhow::Result<u64> {
         let data = api.get_lc_finality_update().await?;
         let update: Value = serde_json::from_str(&data)?;
         let update: LightClientFinalityUpdateCapella =
             serde_json::from_value(update["data"].clone())?;
         info!(
-            "Got lc finality update for slot {:?}",
-            update.attested_header.beacon.slot
+            finalized_slot = %update.finalized_header.beacon.slot,
+            "Got LightClientFinalityUpdate",
         );
+
+        let new_finalized_slot = update.finalized_header.beacon.slot;
+
+        match new_finalized_slot.cmp(&finalized_slot) {
+            Ordering::Equal => {
+                // We already gossiped the latest finality updated with the same finalized slot, no need to serve it again.
+                return Ok(finalized_slot);
+            }
+            Ordering::Less => {
+                // if we panic here, it is a bug
+                bail!("System clock went backwards: Expected finalized slot is less than known finalized slot");
+            }
+            Ordering::Greater => {
+                // Continue
+            }
+        }
+
         let content_key = BeaconContentKey::LightClientFinalityUpdate(
-            LightClientFinalityUpdateKey::new(update.signature_slot),
+            LightClientFinalityUpdateKey::new(new_finalized_slot),
         );
         let content_value = BeaconContentValue::LightClientFinalityUpdate(update.into());
 
-        Self::gossip_beacon_content(portal_clients, content_key, content_value).await
+        Self::gossip_beacon_content(portal_clients, content_key, content_value)
+            .await
+            .map(|()| new_finalized_slot)
```

Comment:
Hm, now that I look at this pattern (of mapping that drops the old value) with fresh eyes it seems a little off. (I think I did it before).

This looks like a cleaner approach:
```suggestion
        Self::gossip_beacon_content(portal_clients, content_key, content_value).await?
        Ok(new_finalized_slot)
```